File: examples-sdk/go/101b-multi-pipeline-benchmark/main.go
// 101b-multi-pipeline-benchmark — Go SDK equivalent
// Compile: vil compile --from go --input 101b-multi-pipeline-benchmark/main.go --release
package main

import vil "github.com/OceanOS-id/vil-go"

func main() {
	p := vil.NewPipeline("MultiPipelineBench", 3090)

	p.Sink(vil.SinkOpts{Name: "gateway", Port: 3090, Path: "/trigger"})
	p.Source(vil.SourceOpts{Name: "l_l_m_upstream", URL: "http://127.0.0.1:4545/v1/chat/completions", JSONTap: "choices[0].delta.content"})

	p.Route("sink.trigger_out", "source.trigger_in", "LoanWrite")
	p.Route("source.response_data_out", "sink.response_data_in", "LoanWrite")
	p.Route("source.response_ctrl_out", "sink.response_ctrl_in", "Copy")

	p.Compile()
}

File: examples-sdk/zig/101b-multi-pipeline-benchmark/main.zig
// 101b-multi-pipeline-benchmark — Zig SDK equivalent
// Compile: vil compile --from zig --input 101b-multi-pipeline-benchmark/main.zig --release

const vil = @import("vil");

pub fn main() void {
    var p = vil.Pipeline.init("MultiPipelineBench", 3090);
    p.sink("gateway", 3090, "/trigger");
    p.source("l_l_m_upstream", "http://127.0.0.1:4545/v1/chat/completions");
    p.route("sink.trigger_out", "source.trigger_in", "LoanWrite");
    p.route("source.response_data_out", "sink.response_data_in", "LoanWrite");
    p.route("source.response_ctrl_out", "sink.response_ctrl_in", "Copy");
    p.compile();
}
